extensions/crypto/00_crypto.js

```diff
@@ -414,6 +427,165 @@
+          switch (format) {
+            case "raw": {
+              const hash = normalizedAlgorithm.hash;
+              let length = keyData.byteLength * 8;
+
+              if (length === 0) {
+                throw new DOMException("Key length is zero", "DataError");
+              }
+
+              if (normalizeAlgorithm.length) {
+                if (
+                  normalizedAlgorithm.length > length ||
+                  normalizedAlgorithm.length <= (length - 8)
+                ) {
+                  throw new DOMException(
+                    "Key length is invalid",
+                    "DataError",
+                  );
+                }
+                length = normalizeAlgorithm.length;
+              }
+
+              if (keyUsages.length == 0) {
+                throw new DOMException("Key usage is empty", "SyntaxError");
+              }
+
+              const handle = {};
+              WeakMapPrototypeSet(KEY_STORE, handle, {
+                type: "raw",
+                data: keyData,
+              });
+
+              const algorithm = {
+                name: "HMAC",
+                length,
+                hash,
+              };
+
+              const key = constructKey(
+                "secret",
+                true,
+                usageIntersection(keyUsages, recognisedUsages),
+                algorithm,
+                handle,
+              );
+
+              return key;
+            }
+            // TODO(@littledivy): jwk
+            default:
+              throw new DOMException("Not implemented", "NotSupportedError");
+          }
+        }
+        // TODO(@littledivy): RSASSA-PKCS1-v1_5
+        // TODO(@littledivy): RSA-PSS
+        // TODO(@littledivy): ECDSA
+        default:
+          throw new DOMException("Not implemented", "NotSupportedError");
+      }
+    }
+
+    /**
+    * @param {string} format
+    * @param {CryptoKey} key
+    * @returns {Promise<any>}
+    */
+    // deno-lint-ignore require-await
+    async exportKey(format, key) {
+      webidl.assertBranded(this, SubtleCrypto);
+      const prefix = "Failed to execute 'exportKey' on 'SubtleCrypto'";
+      webidl.requiredArguments(arguments.length, 2, { prefix });
+      format = webidl.converters.KeyFormat(format, {
+        prefix,
+        context: "Argument 1",
+      });
+      key = webidl.converters.CryptoKey(key, {
+        prefix,
+        context: "Argument 2",
+      });
+
+      const handle = key[_handle];
+      const bits = WeakMapPrototypeGet(KEY_STORE, handle);
+
+      switch (key[_algorithm].name) {
+        case "HMAC": {
+          if (bits == null) {
+            throw new DOMException("Key is not available", "OperationError");
+          }
+          switch (format) {
+            case "raw": {
+              if (bits.length % 8 !== 0) {
+                const desiredLength = Math.round(bits.length / 8) * 8;
+                for (let _i = 0; _i < (desiredLength - bits.length); _i++) {
```

Comment:
This looks wrong. You're trying to pad to the next multiple of 8? You're probably looking for this:
```suggestion
                for (let _i = 8 - bits.length % 8; _i > 0; _i--) {
```
Or:
```rust
for (let _i = 7 & (8 - bits.length % 8); _i > 0; _i--) {
```
That lets you get rid of the `if (bits.length % 8 !== 0) {`.

Reply:
Thanks, fixed